runtime/runtime.go

```diff
@@ -3215,23 +3234,23 @@ func validatePublicKey(
 	getLocationRange func() interpreter.LocationRange,
 	publicKeyValue *interpreter.CompositeValue,
 	runtimeInterface Interface,
-) interpreter.BoolValue {
-
+) (interpreter.BoolValue, error) {
```

Comment:
@turbolent Is it fine to leave this returning two values?

Reply:
Sorry, I missed this! I'm unfortunately not quite following, I'll dedicate some time tomorrow to try this and investigate.

Reply:
@tarakby When would decoding the public key fail, and it is not an `invalidInputsError`? Are there really three different outcomes for public key validation, and if so, what are those?

Reply:
The key decoding fails with a non-`invalidInputsError` when there was an unexpected execution error (C layer issue, memory..).

A key validation outputs: 
 - `true, nil` when the bytes represent a valid key and there was no execution error
 - `false, nil` when the bytes do not represent a valid key and there was no execution error ( `invalidInputsError` from `DecodePublicKey` shows that the execution went well but the bytes do not represent any valid key)
 - `false, error` is returned on **master** when an unexpected error occurred. **However**, this [PR](https://github.com/onflow/flow-go/pull/1605) updates this case to panic instead. The PR is not merged yet, it looks like it's blocked by the current PR: https://github.com/onflow/flow-go/pull/1605#issuecomment-983860729

Reply:
Thanks for the explanation @tarakby! In that case it's probably best to only return a single return value, an error, and `nil` means valid, and non-`nil` a user error. If the FVM side then runs into an `invalidInputsError` , which is an implementation error, it should panic (like in the linked open PR), it should not be returned (confusion of an implementation error as a user error)

Reply:
I changed it so `ValidatePublicKey` only returns an error.

I assume this will break the FVM-Cadence integration because the `Interface.ValidatePublicKey` method in `cadence/runtime/interface.Go` changed. Do I need to do anything to resolve that?

Reply:
@robert-e-davidson3 FVM depends on Cadence, so once we release Cadence, we will have to update FVM and adjust the implementation there. There's nothing else to do here on the Cadence side